### PR TITLE
feat: Forbid funding to wealthy users

### PR DESCRIPTION
**Pull request by savudani8**

### CODE REVIEW

File: faucet/src/http.rs [hunk 1]
```diff
@@ -19,7 +19,10 @@ use tokio::time::timeout;
 const HEALTH_DURATION: Duration = Duration::from_millis(5000);
 
 const KV_STORE_NAME: &str = "store";
-const FAUCET_COOLDOWN_HOURS: i64 = 6;
+const FAUCET_COOLDOWN_HOURS: i64 = 12;
```
Comment (sander2) [inline faucet/src/http.rs:22]: why?

File: faucet/src/http.rs [hunk 2]
```diff
@@ -152,24 +155,34 @@ fn has_request_expired(
...
+async fn is_funding_allowed(
+    provider: &PolkaBtcProvider,
+    account_id: AccountId,
     last_request_json: Option<Json<FaucetRequest>>,
     account_type: FundingRequestAccountType,
 ) -> Result<bool, Error> {
-    // We are subtracting FAUCET_COOLDOWN_HOURS from the milliseconds since the unix epoch.
-    // Unless there's a bug in the std lib implementation of Utc::now() or a false reading from the
-    // system clock, unwrap will never panic
+    let balance = provider.get_free_dot_balance_for_id(account_id.clone()).await?;
```
Comment (sander2) [inline faucet/src/http.rs:164]: Shouldn't we also check the locked balance?

File: faucet/src/http.rs (hunk 1 above)
Comment (sander2) [reply faucet/src/http.rs:22]: Yea but there have been a lot of instances where something went wrong, and users had to wait for the cooldown before they got the vault up and running

File: faucet/src/http.rs [hunk 3]
```diff
@@ -478,7 +547,7 @@ mod tests {
         let (client, tmp_dir) = default_provider_client(AccountKeyring::Alice).await;
         set_exchange_rate(client.clone()).await;
 
-        let bob_account_id: AccountId = AccountKeyring::Bob.to_account_id();
+        let bob_account_id: AccountId = [3; 32].into();
```
Comment (gregdhill) [inline faucet/src/http.rs:550]: What's the reason for this change?

File: faucet/src/http.rs [hunk 4]
```diff
@@ -532,6 +601,8 @@ mod tests {
         let user_allowance_dot: u128 = 1;
         let vault_allowance_dot: u128 = 500;
         let staked_relayer_allowance_dot: u128 = 500;
+        let one_dot: u128 = 10_000_000_000;
```
Comment (gregdhill) [inline faucet/src/http.rs:604]:
```suggestion
        let one_dot: u128 = 10 ** 10;
```

File: faucet/src/http.rs [hunk 5]
```diff
@@ -152,24 +155,39 @@ fn has_request_expired(
...
 }
 
-fn is_funding_allowed(
+async fn is_funding_allowed(
+    provider: &PolkaBtcProvider,
+    account_id: AccountId,
     last_request_json: Option<Json<FaucetRequest>>,
     account_type: FundingRequestAccountType,
 ) -> Result<bool, Error> {
-    // We are subtracting FAUCET_COOLDOWN_HOURS from the milliseconds since the unix epoch.
```
Comment (gregdhill) [inline faucet/src/http.rs:159]: Ideally keep this comment

File: faucet/src/http.rs (hunk 3 above)
Comment (savudani8) [reply faucet/src/http.rs:550]: The Keyring accounts are prefunded and the faucet should not send them even more funds. This change uses an unfunded account to test the faucet

File: faucet/src/http.rs (hunk 4 above)
Comment (savudani8) [reply faucet/src/http.rs]: That doesn't seem to be valid syntax, I used `10u128.pow(10)`

**Review by gregdhill — APPROVED** [review]
